src/classes/BGE_ConfigurationWizard_CTRL.cls

```diff
@@ -200,16 +200,20 @@ public with sharing class BGE_ConfigurationWizard_CTRL {
         };
 
         // looping through the fields on DataImport__c and remove unsupported data types
+        // Remove fields user doesn't have permission to update unless whitelisted
         for (Schema.DescribeFieldResult dfr : allFieldsMap.values()) {
             String dataImportFieldApiName = UTIL_Namespace.StrTokenRemoveNSPrefix(dfr.getName());
 
             String dataImportFieldMappedObject = BDI_DataImportService.getTargetObject(dfr.getName());
 
-            Boolean fieldAllowed = (allowedObjects.contains(dataImportFieldMappedObject)
+            Boolean fieldAllowed = (
+                UTIL_Permissions.canUpdate(dfr, false)
+                && ((allowedObjects.contains(dataImportFieldMappedObject)
                 && allowedDataTypes.contains(dfr.type)
                 && !bannedFields.contains(dataImportFieldApiName)
                 && dfr.getInlineHelpText() != null)
-                || allowedFields.contains(dataImportFieldApiName);
+                || allowedFields.contains(dataImportFieldApiName))
+            );
```

Comment:
Is the call to "BGE_BatchGiftEntry_UTIL.checkFieldPermissions(filteredDFRMap.values());" below on line 223 redundant now, since the field has to be updatable to get in the map in the first place?

Reply:
Yeah, it is redundant as all the fields passed in will always pass. What do you think about the following scenario?

If a user without perms for a field (Campaign Source) accesses a DI Batch that has Campaign Source in its Opportunity Selected Fields... with this update they'll receive an error on the DI Batch page but still be able to click "Edit" and open the Config Wizard. The page will tell them they don't have access to field Campaign Source, the page will not render the New Gift form and not render the Gifts table. They'll still be able to open the Config Wizard, but Campaign Source will not be visible for them in the Opportunity Selected Fields.

Reply:
This is what we discussed earlier today (Monday 4/8) right?  I think this is ok, we may look to providing a more informative blocking message when trying to edit a batch with an inaccessible field, but this is great of this story.

Reply:
And I believe we should still remove the redundant call to "BGE_BatchGiftEntry_UTIL.checkFieldPermissions(filteredDFRMap.values());" on line 223-ish?

Reply:
Yup, exactly. Completely agree, I'll go ahead and remove the call to the BGE util